tests/conftest.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+import threading
+import traceback
+import socket
+
+from server import ConnectionListener
+
+# fmt: off
+motd_dict_test = {
+    "motd": [
+        "- Hello {user_nick}, this is a test MOTD!",
+        "-",
+        "- Foo",
+        "- Bar",
+        "- Baz",
+        "-",
+        "- End test MOTD"
+        ]
+}
+
+# fmt: on
+
+# Based on: https://gist.github.com/sbrugman/59b3535ebcd5aa0e2598293cfa58b6ab#gistcomment-3795790
+@pytest.fixture(scope="function")
+def fail_test_if_there_is_an_error_in_a_thread(monkeypatch):
+    last_exception = None
+
+    class ThreadWrapper(threading.Thread):
+        def run(self):
+            try:
+                super().run()
+            except Exception as e:
+                traceback.print_exc()
+                nonlocal last_exception
+                last_exception = e
+
+    monkeypatch.setattr(threading, "Thread", ThreadWrapper)
+    yield
+    if last_exception:
+        raise last_exception
+
+
+@pytest.fixture()
```

Comment:
It's confusing that this one has `()` at the end while other fixtures don't. It works the same either way, but consistency would be good IMO.

```suggestion
@pytest.fixture
```